### ISSUE TITLE
Filter text-field overflows

### ISSUE THREAD

**Issue by tomivirkki** [vaadin/vaadin-grid#1176]
When the filterable column is narrow enough, the contained text-field overflows the filter box.

Could fix by setting the default text-field's max-width to 100% (in grid styles).

**Comment by samiheikki** [comment]
Do you mean https://github.com/vaadin/vaadin-grid/blob/master/src/vaadin-grid-filter.html#L23 ?
I got blocked by https://github.com/vaadin/vaadin-grid/issues/1252

**Comment by nbabb** [comment]
This still appears to be an issue (see https://vaadin.com/forum/thread/17801517/grid-column-filter-border-is-being-cutoff).  Can we reopen this, or file a new ticket?

**Comment by jouni** [comment]
Another fix for the original issue would be to set `min-width: 0;` for the `<vaadin-text-field>`. That also makes it shrink inside the parent flex container.

**Comment by jouni** [comment]
@nbabb, would it be possible to create a client-side only test case for this? I mean, it would help the team, as they probably don’t have that much experience in running Gradle projects, so it would help to speed things up if they don’t need to spend time on setup.

**Comment by jouni** [comment]
Here’s a starting point. Click "Remix This" to edit it: https://glitch.com/edit/#!/kindly-postbox?path=index.html:1:0

**Comment by jouni** [comment]
Actually, it looks like the problem is that in your app, the `<vaadin-text-field>` is placed inside the grid header cell directly, not wrapped with the `<vaadin-grid-filter>` element. The fix for this issue was applied through the filter element.

**Comment by nbabb** [comment]
Thanks for the replies Jouni.  It seems like Tomi is suggesting a different fix here https://vaadin.com/forum/thread/17801517/grid-column-filter-border-is-being-cutoff  I'm not sure which advice I should take :)  Maybe it would make sense to add any further clarifications to the forum posting?

**Comment by jouni** [comment]
😄 

Sure, we can keep the discussion in the forum post. I think Tomi’s suggestion is the correct workaround for your case. We might still want to do something about it in our components so that others would not stumble on it.
